src/Tutorial.ts

```diff
@@ -0,0 +1,438 @@
+    }
+
+    initiateTutQuickSelect = () : void => {
+        var that = this
+        $("body").on('keydown.tutEventListener',function(e){
+
+            switch(e.which) {
+                case 37: // left
+                    that.tutButtonPrev()
+                break;
+        
+                case 38: // up
+                    that.tutButtonPrev()
+                break;
+        
+                case 39: // right
+                    if(activeTut.getTutorialSteps()[activeTutCurrentStep-1].getType() != TutorialStep.Type.Press){
+                        that.tutButtonNext() 
+                    }
+                break;
+        
+                case 40: // down
+                    that.tutButtonNext()
+                break;
+
+                case 27: //escape
+                e.preventDefault()
+                    that.tutButtonEnd()
+                break;
+        
+                default: return; // exit this handler for other keys
+            }
+        })
+    }
+
+    //cooldown function that prevents too many actions that would cause the tutorial steps to go out of whack
+    startCooldown = () : void => {
+        cooldown = true
+        setTimeout(function(){
+            cooldown = false
+        }, 500)
+    }
+
+    initiateTutStep = (direction:string) :void => {
+        const eagle = Eagle.getInstance()
+
+        var tutStep = activeTut.getTutorialSteps()[activeTutCurrentStep-1]
+
+        //if there is a preFunction set, then we execute it here
+        var preFunction
+
+        if(direction === 'next'){
+            preFunction = tutStep.getPreFunct()
+        }else if(direction === 'prev'){
+            preFunction = tutStep.getBackPreFunct()
+        }
+        
+        if(preFunction != null){
+            preFunction(eagle)
+        }
+        
+        //we always pass through the wait funciton, it is decided there if we actully wait or not
+        this.initiateWaitForElement(tutStep.getWaitType())
+    }
+
+    initiateWaitForElement = (waitType:Wait.Type) :void => {
+        if(waitType===Wait.Type.None){
+            this.pickStepType(activeTut.getTutorialSteps()[activeTutCurrentStep-1],null)
+        }else{
+            //we set a two second timer, the wait will check every .1 seconds for two seconds at which point it is timed out and we abort the tut
+            waitForElementTimer = setInterval(function(){activeTut.waitForElement(waitType)}, 100);  
+            setTimeout(function(){
+                if(waitForElementTimer != null){
+                    clearTimeout(waitForElementTimer);
+                    waitForElementTimer = null;
+                    console.warn('waiting for next tutorial step element timed out')
+                }
+            }, 2000)
+        }
+    }
+
+    waitForElement = (waitType:Wait.Type) :void => {
+        var tutStep = activeTut.getTutorialSteps()[activeTutCurrentStep-1]
+        var elementAvailable = false
+        var selectorElement = tutStep.getSelector()
```

Comment:
Use of selectorElement is a little confusing here. I would expect tutStep.getSelector() to return a string. But the result is called 'selectorElement', so I think it is a HTML element. But within this function, selectorElement is sometimes called like a function, and sometimes not.

Declare the type of the variable. Use let?